pkg/controller/factory/base_controller.go

```diff
@@ -62,6 +65,32 @@ func (s *scheduledJob) Run() {
 	s.queue.Add(DefaultQueueKey)
 }
 
+func waitForPreconditions(controllerName string, stopCh <-chan struct{}, preconditions ...PreconditionFn) error {
+	if len(preconditions) == 0 {
+		return nil
+	}
+	err := wait.PollImmediateUntil(100*time.Millisecond,
```

Comment:
worth to externalize the interval too? I can imagine that something here is expensive.

Reply:
I reworked this a little.